### ISSUE TITLE
`rectangle` function for `skimage.draw`

### ISSUE THREAD

**Issue by TheShadow29** [scikit-image/scikit-image#2688]
Currently the  [skimage.draw](http://scikit-image.org/docs/dev/api/skimage.draw.html) doesn't support drawing a rectangle directly. To do so one has to input all 4 points in the polygon function. 

For most purposes for saving bounding boxes ((x1,y1), width, height) is required. Example [cv2](https://stackoverflow.com/questions/23720875/how-to-draw-a-rectangle-around-a-region-of-interest-in-python).

While such a functionality is available in [matplotlib.patches.Rectangle](https://matplotlib.org/api/patches_api.html), it adds the rectangle to the figure and not the image.

**Comment by TheShadow29** [comment]
If no one is working on this, I would like to do so. The easiest way seems to directly use the Polygon function after getting the 4 coordinates of the rectangle? Or perhaps directly use the line function?

**Comment by soupault** [comment]
Although I understand the popularity of `((x1, y1), width, height)` format, I think that it is quite simple to build something on top of http://scikit-image.org/docs/dev/api/skimage.draw.html#skimage.draw.polygon_perimeter . Above said, I'd keep the implementation of this conversion up to the user.

**Comment by 0x00b1** [comment]
I needed this today. Enjoy!

```Python
def rectangle(r0, c0, width, height):    
    rr, cc = [r0, width, width, r0], [c0, c0, height, height]

    return skimage.draw.polygon(rr, cc)


def rectangle_perimeter(r0, c0, width, height, shape=None, clip=False):    
    rr, cc = [r0, width, width, r0], [c0, c0, height, height]

    return skimage.draw.polygon_perimeter(rr, cc, shape=shape, clip=clip)
```

**Comment by stefanv** [comment]
Let's implement this, but in N-dimensions.  The signature would be `draw_something(origin, shape)` or similar.

**Comment by TheShadow29** [comment]
@stefanv can you elaborate a bit more on what the function should exactly do? I am not sure what drawing in N-dimensions exactly means. Also should the origin be the center of the shape or one corner of the shape?

**Comment by stefanv** [comment]
Also see: https://github.com/scikit-image/scikit-image/pull/2719

**Comment by scotthuang1989** [comment]
@0x00b1  I think It's better add a linewidth parameter, because some time line with 1 pixel width is hardly see.

**Comment by stefanv** [comment]
For drawing boxes, width is not too tricky, since you can do what you showed here: replicate the line.  In N-dimensions, the same trick will work, but you have to repeat in more directions.

**Comment by scotthuang1989** [comment]
ok, got it, I am new to his lib (a couple of hours), I am just used to use opencv's out of box method, it take me a while to figure out whey the box is not shown, maybe other new users will have same issue.

**Comment by soupault** [comment]
Closed via #2719
